### ISSUE TITLE
[FEA] Accept DLPack PyCapsules in .fit() method

### ISSUE THREAD

**Issue by miguelusque** [dmlc/xgboost#6663]
Hi!

I think it would be very useful if XGBoost´s `.fit()` method would accept DLPack PyCapsules as features and labels.

The current behaviour makes the following example to fail:

```
features = jnp.array([[1, 2], [3, 4]])
labels = jnp.array([5, 6])
model = xgb.XGBRegressor(
    gamma=1,                 
    learning_rate=0.01,
    max_depth=1,
    n_estimators=100,                                                                    
    subsample=0.8,
    random_state=34,
    tree_method='gpu_hist'
) 

model.fit(jax.dlpack.to_dlpack(features), jax.dlpack.to_dlpack(labels))
```

I am aware I may use the following instead, and that it would imply zero-copy in GPU memory, but I would like to avoid to use a third library to train an XGBoost model from JAX DeviceArrays.

`model.fit(cp.fromDlpack(jax.dlpack.to_dlpack(features)), cp.fromDlpack(jax.dlpack.to_dlpack(labels)))`

Thank you for having the feature request above into consideration.

Regards,
Miguel

**Comment by trivialfis** [comment]
Would you like to open a PR, I can point you to the code and help answering questions.

**Comment by trivialfis** [comment]
I think even if we implement it we might use cupy as a bridge.  We don't want to have too many dependencies for a machine learning library, especially considering the difficulty of maintaining the CI.
